client/browser/src/shared/components/CodeViewToolbar.tsx

```diff
@@ -102,7 +102,9 @@ export class CodeViewToolbar extends React.Component<CodeViewToolbarProps, CodeV
                         />
                     </li>
                 )}{' '}
-                {!this.props.baseCommitID && (
+                {// Only show the "View file" button if we were able to fetch the file contents
+                // from the Sourcegraph instance
+                !this.props.baseCommitID && this.props.content !== undefined && (
```

Comment:
```suggestion
                !this.props.baseCommitID && (this.props.content !== undefined || this.props.baseContent !== undefined) && (
```
I think this also needs to check `baseContent` because a new added file doesn't have a base content.

Reply:
Are you sure? This line starts with `!this.props.baseCommitID`, and if there's no `baseCommitID`, there will be no `baseContent` (and yes, this should be [enforced by the type system](https://github.com/sourcegraph/sourcegraph/issues/3107#event-2244054897)).

And even if a newly added file doesn't have a `baseContent`, it will still have a `content`

Reply:
I think there is currently code that always sets `baseCommitID` without checking the change type